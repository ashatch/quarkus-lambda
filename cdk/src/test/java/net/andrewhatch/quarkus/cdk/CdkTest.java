package net.andrewhatch.quarkus.cdk;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import software.amazon.awscdk.core.App;
import software.amazon.awscdk.core.AppProps;
import software.amazon.awscdk.core.StackProps;
import software.amazon.awscdk.cxapi.CloudAssembly;
import software.amazon.awscdk.cxapi.CloudFormationStackArtifact;

import java.util.HashMap;
import java.util.Map;

import au.com.origin.snapshots.SnapshotMatcher;
import au.com.origin.snapshots.junit5.SnapshotExtension;

@ExtendWith(SnapshotExtension.class)
public class CdkTest {
    @Test
    public void testStack() {
        final Map<String, Object> context = new HashMap<>();
        context.put(CdkStack.LAMBDA_JAR_PATH, "src/test/resources/lambda-jar-dir");

        final App app = new App(
            AppProps.builder()
            .context(context).build());

        final CdkStack stack = new CdkStack(app, "test", StackProps.builder().build());

        final CloudAssembly synthedApp = app.synth();
        final CloudFormationStackArtifact stackArtifact = synthedApp.getStackArtifact(stack.getArtifactId());
        final Map template = (Map)stackArtifact.getTemplate();

        SnapshotMatcher.expect(template).toMatchSnapshot();
    }
}
