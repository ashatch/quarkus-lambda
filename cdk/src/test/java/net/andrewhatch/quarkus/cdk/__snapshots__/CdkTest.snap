com.myorg.CdkTest.testStack=[
  {
    "Parameters": {
      "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bArtifactHash916FD55F": {
        "Description": "Artifact hash for asset \"46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0b\"",
        "Type": "String"
      },
      "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3Bucket7C43A93C": {
        "Description": "S3 bucket for asset \"46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0b\"",
        "Type": "String"
      },
      "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3VersionKey8B1568C2": {
        "Description": "S3 key for asset version \"46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0b\"",
        "Type": "String"
      }
    },
    "Resources": {
      "quarkuslambda8FEF34C5": {
        "DependsOn": [
          "quarkuslambdaServiceRole1F38A68C"
        ],
        "Properties": {
          "Code": {
            "S3Bucket": {
              "Ref": "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3Bucket7C43A93C"
            },
            "S3Key": {
              "Fn::Join": [
                "",
                [
                  {
                    "Fn::Select": [
                      0,
                      {
                        "Fn::Split": [
                          "||",
                          {
                            "Ref": "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3VersionKey8B1568C2"
                          }
                        ]
                      }
                    ]
                  },
                  {
                    "Fn::Select": [
                      1,
                      {
                        "Fn::Split": [
                          "||",
                          {
                            "Ref": "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3VersionKey8B1568C2"
                          }
                        ]
                      }
                    ]
                  }
                ]
              ]
            }
          },
          "FunctionName": "dev-quarkus-lambda",
          "Handler": "test",
          "MemorySize": 512,
          "Role": {
            "Fn::GetAtt": [
              "quarkuslambdaServiceRole1F38A68C",
              "Arn"
            ]
          },
          "Runtime": "java8",
          "Timeout": 20
        },
        "Type": "AWS::Lambda::Function"
      },
      "quarkuslambdaServiceRole1F38A68C": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": "lambda.amazonaws.com"
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
                ]
              ]
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      }
    }
  }
]


net.andrewhatch.quarkus.cdk.CdkTest.testStack=[
  {
    "Parameters": {
      "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bArtifactHash916FD55F": {
        "Description": "Artifact hash for asset \"46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0b\"",
        "Type": "String"
      },
      "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3Bucket7C43A93C": {
        "Description": "S3 bucket for asset \"46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0b\"",
        "Type": "String"
      },
      "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3VersionKey8B1568C2": {
        "Description": "S3 key for asset version \"46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0b\"",
        "Type": "String"
      }
    },
    "Resources": {
      "quarkuslambda8FEF34C5": {
        "DependsOn": [
          "quarkuslambdaServiceRole1F38A68C"
        ],
        "Properties": {
          "Code": {
            "S3Bucket": {
              "Ref": "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3Bucket7C43A93C"
            },
            "S3Key": {
              "Fn::Join": [
                "",
                [
                  {
                    "Fn::Select": [
                      0,
                      {
                        "Fn::Split": [
                          "||",
                          {
                            "Ref": "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3VersionKey8B1568C2"
                          }
                        ]
                      }
                    ]
                  },
                  {
                    "Fn::Select": [
                      1,
                      {
                        "Fn::Split": [
                          "||",
                          {
                            "Ref": "AssetParameters46c0277996190033243e7490c8ab35b0018660b0fd1a060f13e8c65ad635dd0bS3VersionKey8B1568C2"
                          }
                        ]
                      }
                    ]
                  }
                ]
              ]
            }
          },
          "FunctionName": "dev-quarkus-lambda",
          "Handler": "test",
          "MemorySize": 512,
          "Role": {
            "Fn::GetAtt": [
              "quarkuslambdaServiceRole1F38A68C",
              "Arn"
            ]
          },
          "Runtime": "java8",
          "Timeout": 20
        },
        "Type": "AWS::Lambda::Function"
      },
      "quarkuslambdaServiceRole1F38A68C": {
        "Properties": {
          "AssumeRolePolicyDocument": {
            "Statement": [
              {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": {
                  "Service": "lambda.amazonaws.com"
                }
              }
            ],
            "Version": "2012-10-17"
          },
          "ManagedPolicyArns": [
            {
              "Fn::Join": [
                "",
                [
                  "arn:",
                  {
                    "Ref": "AWS::Partition"
                  },
                  ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
                ]
              ]
            }
          ]
        },
        "Type": "AWS::IAM::Role"
      }
    }
  }
]